Code/GraphMol/MolPickler.cpp

```diff
@@ -1882,27 +1882,17 @@ template <typename T>
 void MolPickler::_pickleSSSR(std::ostream &ss, const RingInfo *ringInfo,
                              std::map<int, int> &atomIdxMap) {
   PRECONDITION(ringInfo, "missing ring info");
-  T tmpT;
-  tmpT = ringInfo->numRings();
-  streamWrite(ss, tmpT);
+  std::uint32_t nrings = ringInfo->numRings();
+  streamWrite(ss, nrings);
   for (unsigned int i = 0; i < ringInfo->numRings(); i++) {
     INT_VECT ring;
     ring = ringInfo->atomRings()[i];
-    tmpT = static_cast<T>(ring.size());
+    T tmpT = static_cast<T>(ring.size());
```

Comment:
Should this be uint32_t as well?

Reply:
If I'm reading this right, this is saying no single ring can be > 255 atoms, but the pickle format supports rings > 255 so I'm a bit confused.

Reply:
> If I'm reading this right, this is saying no single ring can be > 255 atoms, but the pickle format supports rings > 255 so I'm a bit confused.

The type of T is determined by the number of atoms in the molecule. So the only time T would be char instead of uint32_t is when the molecule has <255 atoms. If you have less than 255 atoms then you can't end up with a ring of size>255. :-)

Reply:
Fair enough, I wonder what the equation is for N Vertices versus m rings is.

Reply:
> Fair enough, I wonder what the equation is for N Vertices versus m rings is.

Fixing the problem that you can have more rings than atoms is the general point of the PR, which changes the type of the number of rings pickle element to an int even if the size of the atom index (or number of atoms) is a char. 

This particular line of code is connected to the maximum size of any given ring, and *that* is the number of atoms.